src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c

```diff
@@ -152,9 +183,294 @@ const char *__wrap_sqlite3_errmsg(){
+    expect_string(__wrap__mterror, formatted_msg, "unknown pkg_relation 6");
+
+    int ret = pkg_version_relate(version_a, 6, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 0);
+}
+
+void test_pkg_version_relate_epoch_GT(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 1;
+    version_a->version = "";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_GT, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 1);
+}
+
+void test_pkg_version_relate_epoch_GE(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 1;
+    version_a->version = "";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_GE, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 1);
+}
+
+void test_pkg_version_relate_epoch_EQ(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 1;
+    version_a->version = "";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_EQ, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 0);
+}
+
+void test_pkg_version_relate_epoch_LE(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 1;
+    version_a->version = "";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_LE, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 0);
+}
+
+void test_pkg_version_relate_epoch_LT(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 1;
+    version_a->version = "";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_LT, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 0);
+}
+
+void test_pkg_version_relate_version_deb(void **state)
+{
+    struct pkg_version *version_a = state[0];
+    struct pkg_version *version_b = state[1];
+
+    version_a->epoch = 0;
+    version_a->version = "5.3.0.45";
+    version_a->revision = "";
+    version_b->epoch = 0;
+    version_b->version = "4.18.1";
+    version_b->revision = "";
+
+    int ret = pkg_version_relate(version_a, PKG_RELATION_GT, version_b, VER_TYPE_DEB);
+    assert_int_equal(ret, 1);
```

Comment:
Could you please do the same with this test and the three tests below?

Reply:
DONE